config/rootfs/chromiumos/scripts/build_board.sh

```diff
@@ -51,58 +51,44 @@ echo "Board ${BOARD} setup"
-echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-octopus/profiles/base/make.defaults
-fi
-
-# rammus doesn't require any fixes
-
-if [ "${BOARD}" == "sarien" ]; then
-echo "Patching sarien specific issues"
-sed ':a;N;$!ba;s/DEPEND="\n\tchromeos-base\/fibocom-firmware\n"/# DEPEND="\n\t# chromeos-base\/fibocom-firmware\n# "/g' -i src/overlays/overlay-sarien/chromeos-base/modemfwd-helpers/modemfwd-helpers-0.0.1.ebuild
-fi
-
-if [ "${BOARD}" == "volteer" ]; then
-echo "Patching volteer specific issues"
-echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-volteer/profiles/base/make.defaults
-fi
-
-if [ "${BOARD}" == "zork" ]; then
-echo "Patching zork specific issues"
-echo 'USE="${USE} -tpm tpm2"' >>src/overlays/overlay-zork/profiles/base/make.defaults
-fi
-
-if [ "${BOARD}" == "grunt" ]; then
-echo "Patching grunt specific issue"
-echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-grunt/profiles/base/make.defaults
-fi
+echo "Patching ${BOARD} specific issues"
+case ${BOARD} in
+    coral)
+    sed ':a;N;$!ba;s/DEPEND="\n\tchromeos-base\/fibocom-firmware\n"/# DEPEND="\n\t# chromeos-base\/fibocom-firmware\n# "/g' -i src/overlays/overlay-coral/chromeos-base/modemfwd-helpers/modemfwd-helpers-0.0.1.ebuild
+    sed -i s,'media-libs/apl-hotword-support','# media-libs/apl-hotword-support', src/overlays/overlay-coral/media-libs/lpe-support-topology/lpe-support-topology-0.0.1.ebuild
+    sed -i s,'USE="${USE} cros_ec"','# USE="${USE} cros_ec"', src/overlays/baseboard-coral/profiles/base/make.defaults
+    ;;
+    dedede)
+    echo 'USE="${USE} -tpm tpm2 cr50_onboard"' >>src/overlays/baseboard-dedede/profiles/base/make.defaults
+    ;;
+    hatch)
+    sed -i 's/EC_BOARDS=()/EC_BOARDS=(hatch)/' src/third_party/chromiumos-overlay/eclass/cros-ec-board.eclass
+    echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-hatch/profiles/base/make.defaults
+    ;;
+    nami)
+    echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-nami/profiles/base/make.defaults
+    ;;
+    octopus)
+    sed -i s,'use fuzzer || die',"#use fuzzer || die", src/third_party/chromiumos-overlay/eclass/cros-ec-board.eclass
+    # Workaround b/244460939 T38487 - octopus missing proper tpm USE flags
+    echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-octopus/profiles/base/make.defaults
+    ;;
+    sarien)
+    sed ':a;N;$!ba;s/DEPEND="\n\tchromeos-base\/fibocom-firmware\n"/# DEPEND="\n\t# chromeos-base\/fibocom-firmware\n# "/g' -i src/overlays/overlay-sarien/chromeos-base/modemfwd-helpers/modemfwd-helpers-0.0.1.ebuild
+    ;;
+    volteer)
+    echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-volteer/profiles/base/make.defaults
+    ;;
+    zork)
+    echo 'USE="${USE} -tpm tpm2"' >>src/overlays/overlay-zork/profiles/base/make.defaults
+    ;;
+    grunt)
+    echo 'USE="${USE} -tpm tpm2"' >>src/overlays/baseboard-grunt/profiles/base/make.defaults
+    ;;
+    *)
+    echo "No issues found for this board"
+    ;;
+esac
```

Comment:
I like the idea of having `switch/case` instead of  the set of ifs. I thought that maybe we could split these into separate files named after the boards and have them run by the `build_board.sh`. 

```
|-build_board.d/
              |_ grunt.sh
              |_ nami.sh
              |_ ocopus.sh
```


On one hand it may be even more clean as we're having here 9 different cases to handle, on the other hand handlers are usually one liners. @nuclearcat What do you think of that?

Reply:
I thought about that, but most of scripts likely will be one-liners, also i might replace those lines by some function for better understanding of similar patches like adding tpm flags.
So it will be something like:
```
volteer)
zork)
grunt)
    add_tpm2_use_flags()
    ;;
```

But probably i will use shell scripts with board name where significant patching required (likely will happen with other arm64 chromebooks and panfrost/mesa).